geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnection.java

```diff
@@ -1231,11 +1233,16 @@ long putSubject(Subject subject, long existingUniqueId) {
     return uniqueId;
   }
 
-  @VisibleForTesting
+  @TestOnly
   protected ClientUserAuths getClientUserAuths() {
     return clientUserAuths;
   }
 
+  @TestOnly
+  protected void setClientUserAuths(ClientUserAuths clientUserAuths) {
+    this.clientUserAuths = clientUserAuths;
```

Comment:
I know this is test only, but do you see any harm in also having this synchronize on clientUserAuthsLock? That way if someone in the future uses it in the product, or a concurrent test is written, we will not have concurrency issues introduced. With your other changes it seems important to all places that set the clientUserAuths field synchronize on clientUserAuthsLock

Reply:
done

Reply:
I would prefer to remove this test-only method, and change the tests to make public calls to get the client auths to be null/not-null.